pkg/testing/integration/program.go

```diff
@@ -1728,6 +1728,37 @@ func (pt *ProgramTester) prepareGoProject(projinfo *engine.Projinfo) error {
 		return err
 	}
 
+	// initialize a go.mod for dependency resolution if one doesn't exist
+	_, err = os.Stat(filepath.Join(cwd, "go.mod"))
+	if err != nil {
+		err = pt.runCommand("go-mod-init", []string{goBin, "mod", "init"}, cwd)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = pt.runCommand("go-mod-tidy", []string{goBin, "mod", "tidy"}, cwd)
+	if err != nil {
+		return err
+	}
+
+	pulumiSdkPath := filepath.Join(gopath, "src", "github.com", "pulumi", "pulumi", "sdk")
+
+	// if we have a local pulumi enlistment, test against that
+	_, err = os.Stat(pulumiSdkPath)
+	if err == nil {
+		editStr := fmt.Sprintf("github.com/pulumi/pulumi/sdk=%s", pulumiSdkPath)
+		err = pt.runCommand("go-mod-edit", []string{goBin, "mod", "edit", "-replace", editStr}, cwd)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
We should generalize this to use the same mechanism we use everywhere else - which is to use the `Dependencies` field to specify dependencies that should use "linked" versions (locally built).  There's code that does this for Dep a few lines further down here.

We should not need to special case `pulumi/pulumi/sdk` here - as we should be able to ensure that all tests correctly include `pulumi/pulumi/sdk` in their `Dependencies` list in this repro (and in other repos, they will need to include other  modules, like `pulumi/pulumi-aws/sdk/go/aws` in the AWS package.

Reply:
Good point. I'll make this change.